app/controllers/sheets_controller.rb

```diff
@@ -35,15 +35,24 @@ def power
   end
 
   def clear
-    @sheets = @sheets.order(:n_ability, :title)
     @sheet_type = 0
-    write_remain(0)
+    reverse_check(@sheet_type)
+    write_remain(@sheet_type)
   end
 
   def hard
-    @sheets = @sheets.order(:h_ability, :title)
     @sheet_type = 1
-    write_remain(1)
+    reverse_check(@sheet_type)
+    write_remain(@sheet_type)
+  end
+
+  def reverse_check(type)
```

Comment:
__[rails_best_practices]__
* move model logic into model (@sheets use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)